### ISSUE TITLE
RippleDrawable 

### ISSUE THREAD

**Issue by mehcode** [facebook/litho#341]
- [x] I have searched [existing issues](https://github.com/facebook/litho/issues) and this is not a duplicate

## Version
```
litho_version = '0.13.1'
```
```gradle
implementation "com.facebook.litho:litho-core:$litho_version"
implementation "com.facebook.litho:litho-widget:$litho_version"
compileOnly "com.facebook.litho:litho-annotations:$litho_version"

kapt "com.facebook.litho:litho-annotations:$litho_version"
kapt "com.facebook.litho:litho-processor:$litho_version"
```

## Issues and Steps to Reproduce
 - Set an unmasked `RippleDrawable` as the background of a clickable component. 
 
## Expected Behavior

![native-view](https://user-images.githubusercontent.com/753919/38703025-002ccd68-3e57-11e8-95cf-53ae6d8398de.png) 

## Actual Behavior

![litho-view](https://user-images.githubusercontent.com/753919/38703035-06387266-3e57-11e8-937e-42e2863177f0.png)

## Link to Code
```kotlin
Row.create(c)
    .heightDip(50f)
    .child(
        Image.create(c)
            .widthDip(30f)
            .heightDip(30f)
            .marginDip(YogaEdge.LEFT, 15f)
            .marginDip(YogaEdge.TOP, 15f)
            .clickHandler(/* ... */)
            .backgroundRes(R.drawable.ripple)
            .drawableRes(R.drawable.ic_add))
```

*ripple.xml*
```xml
<?xml version="1.0" encoding="utf-8"?>
<ripple
    xmlns:android="http://schemas.android.com/apk/res/android"
    android:color="?android:colorControlHighlight">
</ripple>
```

**Comment by muraziz** [comment]
TLDR; Unbounded ripple works only with `MountSpec`s

Basically the ability to draw outside the view bounds is kinda baked into the views. Drawing outside view bounds is possible only when view has background drawable AND drawable's `isProjected()` method is `true`. You can see that line here: http://androidxref.com/7.0.0_r1/xref/frameworks/base/core/java/android/view/View.java#13574. When view is invalidated it checks those two conditions and calls `damageInParent()` which allows to draw outside the bounds. 

Unfortunately that condition is not reachable for litho components as litho component background works differently. When you have a component and you specify background, then background is created as a separate drawable and is customly handled (it's not View's background). Even if you wrap component to view like (`wrapInView`) it will create a `ComponentHos`t which is a `ViewGroup` and setting backround to it won't set backround from `View`'s perspective, rather it is just another drawable like foreground, border which are kept together.

The only case where we actually set `View`'s background when calling `component.background()` is when component is View `MountSpec`. You can try this following code:

```
     CustomViewComponent.create(c)
             .backgroundRes(R.drawable.ripple)
            .build()

__________________________

@MountSpec
public class CustomViewComponentSpec {
  private static final int DEFAULT_SIZE = 250;

  @OnCreateMountContent
  static FrameLayout onCreateMountContent(ComponentContext c) {
    return new FrameLayout(c);
  }

  @OnMeasure
  static void onMeasure(
      ComponentContext c, ComponentLayout layout, int widthSpec, int heightSpec, Size size) {
    if (SizeSpec.getMode(widthSpec) == SizeSpec.UNSPECIFIED &&
        SizeSpec.getMode(heightSpec) == SizeSpec.UNSPECIFIED) {
      size.width = DEFAULT_SIZE;
      size.height = DEFAULT_SIZE;
    } else {
      MeasureUtils.measureWithEqualDimens(widthSpec, heightSpec, size);
    }
  }
}
```
